packages/flutter_tools/lib/src/isolated/native_assets/macos/native_assets_host.dart

```diff
@@ -104,6 +114,126 @@ Future<void> setInstallNameDylib(File dylibFile) async {
+  final ProcessResult installNameResult = await globals.processManager.run(
+    <String>[
+      'otool',
+      '-D',
+      dylibFile.path,
+    ],
+  );
+  if (installNameResult.exitCode != 0) {
+    throwToolExit(
+      'Failed to get the install name of $dylibFile:\n${installNameResult.stderr}',
+    );
+  }
+
+  // The output of `otool -D` looks like below. For each architecture, a
+  // separate install name is reported, which are not necessarily the same.
+  //
+  // /build/native_assets/ios/buz.framework/buz (architecture x86_64):
+  // @rpath/libbuz.dylib
+  // /build/native_assets/ios/buz.framework/buz (architecture arm64):
+  // @rpath/libbuz.dylib
+
+  final Iterator<String> lines =
+    (installNameResult.stdout as String).trim().split('\n').iterator;
+  final Set<String> installNames = <String>{};
+
+  while (lines.moveNext()) {
+    final String line = lines.current;
+    if (line.contains('(architecture')) {
+      continue;
+    }
+    installNames.add(line.trim());
+  }
+
+  return installNames;
+}
+
+Future<List<String>> getDependencyInstallNamesDylib(File dylibFile) async {
+  final ProcessResult listDependenciesResult = await globals.processManager.run(
+    <String>[
+      'otool',
+      '-L',
+      dylibFile.path,
+    ],
+  );
+  if (listDependenciesResult.exitCode != 0) {
+    throwToolExit(
+      'Failed to get the dependencies of $dylibFile:\n${listDependenciesResult.stderr}',
+    );
+  }
+
+  // The output of `otool -L` looks like below. For each architecture, there
+  // is a separate list of dependencies. The first item in the list of
+  // dependencies describes the library itself. The rest are the dependencies.
+  //
+  // /build/native_assets/ios/buz.framework/buz (architecture x86_64):
+  //         @rpath/buz.framework/buz (compatibility version 0.0.0, current version 0.0.0)
+  //         @rpath/libbar.dylib (compatibility version 0.0.0, current version 0.0.0)
+  // /build/native_assets/ios/buz.framework/buz (architecture arm64):
+  //         @rpath/buz.framework/buz (compatibility version 0.0.0, current version 0.0.0)
+  //         @rpath/libbar.dylib (compatibility version 0.0.0, current version 0.0.0)
+
+  final Iterator<String> lines =
+    (listDependenciesResult.stdout as String).trim().split('\n').iterator;
+  final Set<String> dependencies = <String>{};
+  bool isSelf = false;
+
+  while (lines.moveNext()) {
+    final String line = lines.current;
+    if (line.contains('(architecture')) {
+      isSelf = true;
+      continue;
+    }
+    if (isSelf) {
+      isSelf = false;
+      continue;
+    }
+    dependencies.add(line.trim().split(' ')[0]);
+  }
+
+  return dependencies.toList();
+}
+
+Future<void> changeDependencyInstallNameDylib(
+  File dylibFile, {
+  required String from,
+  required String to,
+}) async {
+   final ProcessResult changeDependencyResult = await globals.processManager.run(
+    <String>[
+      'install_name_tool',
+      '-change',
```

Comment:
Is this a param that can be provided multiple times? If so, we should rewrite the function so that it takes a map or a list of tuples. Then we have less process invocations.